charabia/src/segmenter/camel_case.rs

```diff
@@ -66,6 +47,31 @@ impl<'t> Iterator for CamelCaseParts<'t> {
     }
 }
 
+fn find_camel_case_boundary(str: &str) -> Option<usize> {
+    // CamelCase boundary consists of at least 2 code-points. Avoid iterating over shorter strings.
+    // Note that using `remainder.chars().count() == 1` may catch more cases (non-ASCII strings)
+    // but the main focus here is on " ", "-" and similar that are abundantly produced
+    // by `split_word_bounds()` in the Latin segmenter and mere `len()` performs better at that.
+    if str.len() < 2 {
+        return None;
+    }
+
+    let mut last_char_was_lowercase = false;
+
+    str.find(|c: char| {
+        if c.is_mark_nonspacing() {
+            return false;
+        }
```

Comment:
There will be 2 concurrent ways to check if a character is a nonspacing mark, see the normalizer below:
https://github.com/meilisearch/charabia/blob/main/charabia/src/normalizer/nonspacing_mark.rs#L36-L39

this isn't a big issue, but I'd prefer having 1 way to do it. I don't exclude the library you are using and it could probably replace the dictionary in the normalizer, I put below the related PR showing contributor motivations:
https://github.com/meilisearch/charabia/issues/137
https://github.com/meilisearch/charabia/pull/146

Reply:
Aha, good catch. Looking into the linked issue and PR it seems that the `finl_unicode` or other crates dedicated to finding Char's category were not considered. If this PR gets merged we'll need to check for more than `Mn` category anyway so I suppose we'll want to use a library.

I'm all up for measuring performance of the various crates and (the current HashSet implementation) to make an educated decision what approach to choose here. Though it would probably be best done in a separate PR?

Reply:
@goodhoko, I agree with your approach of separating it into another PR.
Let's consider this as resolved